app/modules/users/components/profile/profile-details.section.tsx

```diff
@@ -46,6 +53,52 @@ export const ProfileDetailsSection: FC<{
       )}
       render={({ isEdit, setIsEdit }) => (
         <div className="flex flex-col gap-8">
+          <div className="flex items-start gap-4">
+            {nameLabel}
+            {isEdit ? (
+              <Form
+                onSubmit={() => setIsEdit(false)}
+                className="flex items-start gap-4 grow"
+                method="put"
+              >
+                <input type="hidden" name="fields" value="firstName" />
+                <input type="hidden" name="firstName" value={firstName ?? ''} />
+
+                <TextField
+                  label="First Name"
+                  size="small"
+                  variant="outlined"
+                  defaultValue={firstName}
+                  onChange={({ target }) => setFirstName(target.value)}
+                />
```

Comment:
`<input type="hidden" name="firstName" value={firstName ?? ''} />` is not needed, remix allows you to use uncontrolled inputs, so you don't need `onChange` at all as well

```suggestion
                <TextField
                  label="First Name"
                  size="small"
                  variant="outlined"
                  defaultValue={user.profile?.firstName}
                  name="firstName"
                />
```